test/nicos_ess/loki/test_loki_scriptbuilder_model.py

```diff
@@ -0,0 +1,183 @@
+#  -*- coding: utf-8 -*-
+# *****************************************************************************
+# NICOS, the Networked Instrument Control System of the MLZ
+# Copyright (c) 2009-2021 by the NICOS contributors (see AUTHORS)
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation; either version 2 of the License, or (at your option) any later
+# version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+# Module authors:
+#
+#   Ebad Kamil <Ebad.Kamil@ess.eu>
+#   Matt Clarke <matt.clarke@ess.eu>
+#
+# *****************************************************************************
+
+from nicos_ess.loki.gui.loki_scriptbuilder_model import LokiScriptModel
+
+
+HEADERS = ['COLUMN_1', 'COLUMN_2', 'COLUMN_3']
+
+
+def create_loki_script_model(num_rows=4, data=None):
+    model = LokiScriptModel(HEADERS, num_rows)
+    if data is not None:
+        model.table_data = data
+    return model
+
+
+def test_initialization_done_correctly():
+    model = create_loki_script_model()
+    # check if initialized with empty data
+    assert not any(data for data in sum(model.table_data, []))
+
+
+def test_row_inserted_at_position():
+    data = [
+        ['00', '01', '02'],
+        ['10', '11', '12'],
+        ['20', '21', '22'],
+        ['30', '31', '32'],
+    ]
+    # Create table with DATA
```

Comment:
```suggestion
    # Create table with data
```
Why are you shouting? ;)